docs/configuration.rst
Configuration
=============

There are two phases of configuration : the first one is related to django and the second one is specific to Picturosaure.

In the ``src/picturosaure`` folder there is file ``local_settings.example.py``. Copy it in the same directory as ``local_settings.py`` :


.. code-block:: bash

    cp src/picturosaure/local_settings.example.py src/picturosaure/local_settings.example.py


Django settings
###############

Please see the `django documentation <https://docs.djangoproject.com/fr/3.0/ref/settings>`_ for extended documentation.

.. attribute:: SECRET_KEY

Default: ``""``

A secret key for a particular Django installation. This is used to provide cryptographic signing, and should be set to a unique, unpredictable value. This value should be kept secret.

.. warning::
    Django can't run without a secret key.

.. attribute:: DEBUG

Default: ``False``

A boolean that turns on/off debug mode. You should use ``DEBUG=False`` for production.

.. attribute:: ALLOWED_HOSTS

Default: ``[]``

A list of strings representing the host/domain names that this Django site can serve.

.. warning::
    Django can't run with ``DEBUG=False`` and ``ALLOWED_HOSTS=[]``.

.. attribute:: DATABASES

Default: ``{
    "default": {
        "ENGINE": "django.db.backends.sqlite3",
        "NAME": BASE_DIR / "db.sqlite3",
    }
}``

If you use a postgresql database, on the same host as where you installed DinoMail, with the above values, it should look like this:

.. code-block:: python

    DATABASES = {
        "default": {
            "ENGINE": "django.db.backends.postgresql",
            "NAME": "dinomail",
            "USER": "dinomail",
            "PASSWORD": "secret",
            "HOST": "localhost",
        }
    }

    
.. attribute:: LANGUAGE_CODE

Default: ``"en-us"``

User will not be able to change the interface language. However, you can select the language you want from the listed below : 

 * English (``'en-us'``)
 * French (``'fr'``)

Those are the languages currently supported for DinoMail.

.. attribute:: TIME_ZONE

Default: ``"UTC"``

Time zone of the server.

.. attribute:: STATIC_ROOT

Default: ``BASE_DIR / "staticfiles"``

Folder in which the static files should be copied. You should make an alias for ``/static`` to this directory. 

.. attribute:: MEDIA_ROOT

Default: ``BASE_DIR / "media"``

Folder in which the media files should be uploaded. You should make an alias for ``/media`` to this directory. 


Picturosaure settings
#####################

There are some settings for picturosaure : 

.. attribute::PICTUROSAURE_USE_WATERMARK

Default : ``True``

If set to False, no watermark is added when the "add watermark" function is executed.

.. attribute::PICTUROSAURE_WATERMARK_POS

Default : ``"bottom-right"``

Position of the watermark. Should be one of the following : 

 * ``"top-left"``
 * ``"top-right"``
 * ``"bottom-left"``
 * ``"bottom-right"``

.. attribute:: PICTUROSAURE_WATERMARK_MARGIN

Default: ``20``

Margin to apply from the selected watermark position.

.. attribute:: PICTUROSAURE_WATERMARK_COLOR

Default : ``(255, 255, 255, 128)``

Color of the watermark in RGBA format (0-255).

.. attribute:: PICTUROSAURE_WATERMARK_SIZE

Default: ``100``

Size of the watermak.

.. attribute:: PICTUROSAURE_WATERMARK_TEXT

Default: ``"©"``

Text to print in the watermark.

.. attribute:: PICTUROSAURE_TITLE

Default: ``"Picturosaure"``

Text displayed in title.

.. attribute:: PICTUROSAURE_CONTACT

Default: ``"example@example.org"``

Contact email. If set to `""`, the mail icon is not displayed.

.. attribute:: PICTUROSAURE_LICENSE

Default: ``"""Where not stated otherwise, the photograph is distributed under 
the <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> license (attribution to )."""``

The text is displayed on the right panel. Is supposed to give the default license of the pictures.

.. attribute:: PICTUROSAURE_LICENSE_NAME

Default: ``""``

Text to display right to the HTML5UP license information. If set to ``""`` nothing is displayed.

CHANGELOG.md
# Version 0.1.1
## Context
**Codename : Caihong juji**

Caihong juji was a beautiful feathered dinosaur.

## Features

Quick fix documentation and media url.

## Update information
No update information.

# Version 0.1.0
## Context
**Codename : Caihong juji**

Caihong juji was a beautiful feathered dinosaur.

## Features

The version 0.1 of Picturosaure includes the following features :

 * Upload pictures and display them in a nice way
 * Link pictures to licenses
 * Add description, location and date to pictures
 * Automatically add watermarks to pictures
 * Add social icons and links

## Update information
There is no update information as there is no previous versions.


src/picturosaure/settings.py
"""
Django settings for picturosaure project.

Generated by 'django-admin startproject' using Django 3.1.

For more information on this file, see
https://docs.djangoproject.com/en/3.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.1/ref/settings/
"""

import sys
from pathlib import Path

BASE_DIR = Path(__file__).resolve(strict=True).parent.parent

SECRET_KEY = ""

DEBUG = False

ALLOWED_HOSTS = []

INSTALLED_APPS = [
    "django.contrib.admin",
    "django.contrib.auth",
    "django.contrib.contenttypes",
    "django.contrib.sessions",
    "django.contrib.messages",
    "django.contrib.staticfiles",
    "core",
]

MIDDLEWARE = [
    "django.middleware.security.SecurityMiddleware",
    "django.contrib.sessions.middleware.SessionMiddleware",
    "django.middleware.common.CommonMiddleware",
    "django.middleware.csrf.CsrfViewMiddleware",
    "django.contrib.auth.middleware.AuthenticationMiddleware",
    "django.contrib.messages.middleware.MessageMiddleware",
    "django.middleware.clickjacking.XFrameOptionsMiddleware",
]

ROOT_URLCONF = "picturosaure.urls"

TEMPLATES = [
    {
        "BACKEND": "django.template.backends.django.DjangoTemplates",
        "DIRS": [],
        "APP_DIRS": True,
        "OPTIONS": {
            "context_processors": [
                "django.template.context_processors.debug",
                "django.template.context_processors.request",
                "django.contrib.auth.context_processors.auth",
                "django.contrib.messages.context_processors.messages",
                "picturosaure.processors.conf_processor",
            ],
        },
    },
]

WSGI_APPLICATION = "picturosaure.wsgi.application"

DATABASES = {
    "default": {
        "ENGINE": "django.db.backends.sqlite3",
        "NAME": BASE_DIR / "db.sqlite3",
    }
}

AUTH_PASSWORD_VALIDATORS = [
    {
        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.CommonPasswordValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",
    },
]


LANGUAGE_CODE = "en-us"

TIME_ZONE = "UTC"

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.1/howto/static-files/

STATIC_URL = "/static/"
MEDIA_URL = "/media/"
STATIC_ROOT = BASE_DIR / "staticfiles"
MEDIA_ROOT = BASE_DIR / "media"

PICTUROSAURE_USE_WATERMARK = True
PICTUROSAURE_WATERMARK_MARGIN = 20
PICTUROSAURE_WATERMARK_COLOR = (255, 2155, 255, 128)
PICTUROSAURE_WATERMARK_SIZE = 100
PICTUROSAURE_WATERMARK_TEXT = "© "
PICTUROSAURE_WATERMARK_POS = "bottom-right"
PICTUROSAURE_TITLE = "Picturosaure"
PICTUROSAURE_CONTACT = "example@example.org"
PICTUROSAURE_LICENSE = """
Where not stated otherwise, the photograph is distributed under the <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> license (attribution to ).
"""
PICTUROSAURE_LICENSE_NAME = ""

try:
    from picturosaure.local_settings import *
except:
    if "test" not in sys.argv or "test_coverage" in sys.argv:
        raise Exception("No local_settings.py file found in picturosaure directory")

if "test" in sys.argv or "test_coverage" in sys.argv:
    DATABASES = {"default": {"ENGINE": "django.db.backends.sqlite3"}}
    SECRET_KEY = "b-nmqgd^$uo-%h_%wrp(07r9(p8q-2)ak5o_hds=2sx-)7py2@"
